re.c

```diff
@@ -1747,15 +1747,19 @@ rb_reg_search_set_match(VALUE re, VALUE str, long pos, int reverse, int set_back
         .range = reverse ? 0 : len,
     };
 
-    VALUE match = match_alloc(rb_cMatch);
-    struct re_registers *regs = RMATCH_REGS(match);
+    struct re_registers regs;
+    memset(&regs, 0, sizeof(struct re_registers));
```

Comment:
You can use `MEMZERO` macro.
```suggestion
    MEMZERO(&regs, 0, struct re_registers);
```
or
```suggestion
    memset(&regs, 0, sizeof(regs));
```
or
```suggestion
   struct re_registers regs = {0}; 
```

Reply:
Okay thanks, I'll use c99 initializer 👍